packages/eslint-plugin-pf-codemods/lib/rules/v5/datalist-remove-selectableRow.js

```diff
@@ -1,25 +1,244 @@
+          toReplace.push([n, sc.getText(n.object)]);
+          handleRightAssignmentSide_Function(n.parent.right, toReplace);
+        } else if (
+          n.type === "AssignmentExpression" &&
+          n.left.name === identifier.name
+        ) {
+          handleRightAssignmentSide_Object(n.right, toReplace);
         }
+      });
+
+      const variableValue = variable?.defs[0].node.init;
+      variableValue &&
+        handleRightAssignmentSide_Object(variableValue, toReplace);
+    };
+
+    const handleRightAssignmentSide_Function = (node, toReplace) => {
+      if (node.type.includes("FunctionExpression")) {
+        swapCallbackParam(node.params, toReplace);
+      } else if (node.type === "Identifier") {
+        handleIdentifierOfFunction(node, toReplace);
       }
     };
+
+    const handleIdentifierOfFunction = (identifier, toReplace) => {
+      const variable = findVariableDeclaration(
+        identifier.name,
+        sc.getScope(identifier)
+      );
+
+      const variableType = variable?.defs[0].type;
+      const variableNode = variable?.defs[0].node;
+
+      if (variableType === "Variable") {
+        const nodesUsingVariable = variable.references.map(
+          (ref) => ref.identifier.parent
+        );
+
+        nodesUsingVariable.forEach((n) => {
+          if (
+            n.type === "AssignmentExpression" &&
+            n.left.name === identifier.name
+          ) {
+            handleRightAssignmentSide_Function(n.right, toReplace);
+          }
+        });
+
+        variableNode.init &&
+          handleRightAssignmentSide_Function(variableNode.init, toReplace);
+      }
+
+      if (variableType === "FunctionName") {
+        swapCallbackParam(variableNode.params, toReplace);
+      }
+    };
+
+    const jsxOpeningsElementsToFix = getAllJSXElements(context)
+      .map((elem) => elem.openingElement)
+      .filter(
+        (openingElem) =>
+          openingElem.name.name === dataListImport.local.name &&
+          openingElem.attributes.some(
+            (attr) =>
+              attr.name.name === "selectableRow" &&
+              attr.value?.type === "JSXExpressionContainer"
+          )
+      );
+
+    let toReplace = [];
+
+    for (const elem of jsxOpeningsElementsToFix) {
+      const selectableRowAttr = elem.attributes.find(
+        (attr) =>
+          attr.name?.name === "selectableRow" &&
+          attr.value?.type === "JSXExpressionContainer"
+      );
+
+      if (!selectableRowAttr) {
+        continue;
+      }
+
+      const expr = selectableRowAttr.value.expression;
+
+      if (expr.type === "ObjectExpression") {
+        handleObjectExpression(expr, toReplace);
+      }
+
+      if (expr.type === "Identifier") {
+        handleIdentifierOfObject(expr, toReplace);
+      }
+
+      toReplace.push([selectableRowAttr.name, "onSelectableRowChange"]);
+    }
+
+    // Having the same fix more than once would lead to errors
+    let toReplaceUnique = [];
```

Comment:
(think I accidentally deleted the wrong thing)

Are there any particular fixes that are being duplicated within toReplace? Just wondering if needing to have another array just to get the unique toReplace items means the fix for the original issue is too complicated or anything.

Reply:
I am testing it on this file:

```
import * as React from "react";
import { DataList } from "@patternfly/react-core";

const onChange = (id) => {};
function onChangeFunc(id, event) {}

const selectableRowObject = { onChange: function (id, event) {} };
selectableRowObject.onChange = onChangeFunc;
selectableRowObject["onChange"] = onChange;

let selectableRowObject2 = { onChange: onChangeFunc };
selectableRowObject2 = selectableRowObject;

<>
  <DataList selectableRow={{ onChange: (id, event) => {} }} />
  <DataList selectableRow={{ onChange: onChangeFunc }} />
  <DataList selectableRow={{ onChange }} />

  <DataList selectableRow={selectableRowObject} />
  <DataList selectableRow={selectableRowObject2} />
</>;
```

Problem is when `onChangeFunc` gets referenced more than once. If you comment out this part:
```
<DataList selectableRow={selectableRowObject} />
<DataList selectableRow={selectableRowObject2} />
```
It will work without the need of `toReplaceUnique`, otherwise it breaks (don't really remember why right now, but I guess when attempting to replace a token which has already been replaced).